exchanges/okx/okx_wrapper.go

```diff
@@ -1072,89 +1053,87 @@ func (ok *Okx) GetOrderHistory(ctx context.Context, req *order.GetOrdersRequest)
-			},
-		})
-		if er != nil {
-			return nil, er
+		if orderStatus == order.New {
+			continue
 		}
-		for i := range response {
-			orderSide := response[i].Side
-			var orderStatus order.Status
-			if strings.EqualFold(response[i].State, "canceled") || strings.EqualFold(response[i].State, "cancelled") {
-				orderStatus = order.Cancelled
-			} else {
-				orderStatus, er = order.StringToOrderStatus(strings.ToUpper(response[i].State))
-				if er != nil {
-					log.Errorf(log.ExchangeSys, "%s %v", ok.Name, er)
-				}
-			}
-			if orderStatus == order.New {
-				continue
-			}
-			pair, er := ok.GetPairFromInstrumentID(response[i].InstrumentID)
-			if er != nil {
-				log.Errorf(log.ExchangeSys, "%s %v", ok.Name, er)
-				continue
-			}
-			if len(req.Pairs) == 0 || len(req.Pairs) == 1 && req.Pairs[0].Equal(currency.EMPTYPAIR) {
-				goto createDetail
-			} else {
-				for x := range req.Pairs {
-					if req.Pairs[x].Equal(pair) {
-						goto createDetail
-					}
-				}
-			}
+		pair, err := ok.GetPairFromInstrumentID(response[i].InstrumentID)
+		if err != nil {
+			log.Errorf(log.ExchangeSys, "%s %v", ok.Name, err)
 			continue
-		createDetail:
-			var oType order.Type
-			switch strings.ToUpper(response[i].OrderType) {
-			case OkxOrderMarket:
-				oType = order.Market
-			case OkxOrderLimit:
-				oType = order.Limit
-			case OkxOrderPostOnly:
-				oType = order.PostOnly
-			case OkxOrderFOK:
-				oType = order.FillOrKill
-			case OkxOrderIOC:
-				oType = order.IOS
-			case OkxOrderOptimalLimitIOC:
-				oType = order.UnknownType
+		}
+		if len(req.Pairs) == 0 || len(req.Pairs) == 1 && req.Pairs[0].Equal(currency.EMPTYPAIR) {
+			goto createDetail
+		} else {
+			for x := range req.Pairs {
+				if req.Pairs[x].Equal(pair) {
+					goto createDetail
+				}
 			}
-			orders = append(orders, order.Detail{
-				Amount:          response[i].Size,
-				Pair:            pair,
-				Price:           response[i].Price,
-				ExecutedAmount:  response[i].LastFilledSize,
-				RemainingAmount: response[i].Size - response[i].LastFilledSize,
-				Fee:             response[i].FeeCurrency,
-				Exchange:        ok.Name,
-				OrderID:         response[i].OrderID,
-				ClientOrderID:   response[i].ClientSupplierOrderID,
-				Type:            oType,
-				Side:            orderSide,
-				Status:          orderStatus,
-				AssetType:       req.AssetType,
-				Date:            response[i].CreationTime,
-				LastUpdated:     response[i].UpdateTime,
-			})
 		}
-	default:
-		return nil, errInvalidInstrumentType
+		continue
+	createDetail:
+		var oType order.Type
+		switch strings.ToUpper(response[i].OrderType) {
+		case OkxOrderMarket:
+			oType = order.Market
+		case OkxOrderLimit:
+			oType = order.Limit
+		case OkxOrderPostOnly:
+			oType = order.PostOnly
+		case OkxOrderFOK:
+			oType = order.FillOrKill
+		case OkxOrderIOC:
+			oType = order.IOS
+		case OkxOrderOptimalLimitIOC:
+			oType = order.UnknownType
+		}
+		orders[i] = order.Detail{
```

Comment:
This block is used a few times we could probably make this a method which returns an order type.